### ISSUE TITLE
Cannot use slot props without using let in the live view

### ISSUE THREAD

**Issue by mathewdgardner** [msaraiva/surface#64]
We get a `BadArityError` when a parent component uses `:props={{ assigns }}` on the slot or calls `@inner_content.(assigns)` in its render function  and the live view does not use `:let` on the parent.

I think this is related to #2 in that I am trying to pass on props from a parent to a child. The only way I can seem to get it to work is if the live view author knows about these props in advance and uses the `:let` directive on the parent. This is problematic since some props are optional and requires the live view author to have intimate knowledge of some implementation details of each component. I'm trying to avoid this since I am working on creating a stylized component library with built-in components.

Parent component passing props on:

```elixir
defmodule Parent do
  use Surface.Component

  slot default, required: true, props: [:foo]

  def render(assigns) do
    ~H"""
    <div>
      <slot :props={{ foo: "bar" }} />
    </div>
    """
  end
end
```

Child component with a prop:

```elixir
defmodule Child do
  use Surface.Component

  property foo, :string

  def render(assigns) do
    ~H"""
    <div>{{ @foo }}</div>
    """
  end
end
```

Live view without using `let`:

```elixir
defmodule ExampleWeb.PropsLive do
  use Surface.LiveView

  def render(assigns) do
    ~H"""
      <Parent>
        <Child />
      </Parent>
    """
  end
end
```

The error we get:

```
** (exit) an exception was raised:
    ** (BadArityError) #Function<2.26325794/0 in Surface.ContentHandler.default_content_fun/3> with arity 0 called with 1 argument ([foo: "bar"])
        (surface_antd 0.1.0) lib/parent.ex:9: Parent."render (overridable 1)"/1
        (phoenix_live_view 0.8.1) lib/phoenix_live_view/utils.ex:332: Phoenix.LiveView.Utils.render_view/2
        (phoenix_live_view 0.8.1) lib/phoenix_live_view/utils.ex:91: Phoenix.LiveView.Utils.to_rendered/2
        (phoenix_live_view 0.8.1) lib/phoenix_live_view/diff.ex:233: Phoenix.LiveView.Diff.traverse/5
        (phoenix_live_view 0.8.1) lib/phoenix_live_view/diff.ex:288: anonymous fn/3 in Phoenix.LiveView.Diff.traverse_dynamic/5
        (elixir 1.10.2) lib/enum.ex:2111: Enum."-reduce/3-lists^foldl/2-0-"/3
        (phoenix_live_view 0.8.1) lib/phoenix_live_view/diff.ex:221: Phoenix.LiveView.Diff.traverse/5
        (phoenix_live_view 0.8.1) lib/phoenix_live_view/diff.ex:92: Phoenix.LiveView.Diff.render/3
        (phoenix_live_view 0.8.1) lib/phoenix_live_view/static.ex:275: Phoenix.LiveView.Static.to_rendered_content_tag/4
        (phoenix_live_view 0.8.1) lib/phoenix_live_view/static.ex:137: Phoenix.LiveView.Static.render/3
        (phoenix_live_view 0.8.1) lib/phoenix_live_view/controller.ex:35: Phoenix.LiveView.Controller.live_render/3
        (phoenix 1.4.14) lib/phoenix/router.ex:288: Phoenix.Router.__call__/2
        (example 0.1.0) lib/example_web/endpoint.ex:1: ExampleWeb.Endpoint.plug_builder_call/2
        (example 0.1.0) lib/plug/debugger.ex:130: ExampleWeb.Endpoint."call (overridable 3)"/2
        (example 0.1.0) lib/example_web/endpoint.ex:1: ExampleWeb.Endpoint.call/2
        (phoenix 1.4.14) lib/phoenix/endpoint/cowboy2_handler.ex:42: Phoenix.Endpoint.Cowboy2Handler.init/4
        (cowboy 2.7.0) /Users/mathew/example/deps/cowboy/src/cowboy_handler.erl:41: :cowboy_handler.execute/2
        (cowboy 2.7.0) /Users/mathew/example/deps/cowboy/src/cowboy_stream_h.erl:320: :cowboy_stream_h.execute/3
        (cowboy 2.7.0) /Users/mathew/example/deps/cowboy/src/cowboy_stream_h.erl:302: :cowboy_stream_h.request_process/3
        (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
```

Live view using `:let` directive on the parent and passing prop to child:

```elixir
defmodule ExampleWeb.PropsLive do
  use Surface.LiveView

  def render(assigns) do
    ~H"""
      <Parent :let={{ foo: foo }}>
        <Child foo={{ foo }} />
      </Parent>
    """
  end
end
```

The above example works but what I'm really trying to do, though, and why I ran into this is to have dynamic props. The dream is to have the child's properties initialized by the parent passing them on and using `:let` be optional since some or all may not need to be set by the user. I realize that this maybe a slightly different issue and that dynamic props as a feature are still a work in progress.

With guidance, I could take a crack at a PR to either make using `:let` optional or even dynamic props since I think the dynamic props implementation may also address this issue.

**Comment by msaraiva** [comment]
Hi @mathewdgardner!

Thanks for reporting this issue. I'll have to drop support for `inner_content.()` in order to fix this, i.e. we'll always gonna need to pass an argument, even if we don't have any props. Like:

```
{{ @inner_content.([]) }}
```

This is no big deal since it's the standard way to do it in LiveView and, in most cases now, we'll be using `<slot/>` instead.

I'll fix it tonight. Thanks!

**Comment by msaraiva** [comment]
Fixed in d55ace4c38216b1d7f9a04dc982e5b0d7e282195.
